testing/src/main/java/org/oppia/android/testing/espresso/TextInputAction.kt

```diff
@@ -0,0 +1,56 @@
+package org.oppia.android.testing.espresso
+
+import android.view.View
+import androidx.test.espresso.matcher.BoundedMatcher
+import com.google.android.material.textfield.TextInputLayout
+import org.hamcrest.Description
+
+/**
+ * Action for matching error text and performing other actions on TextInputLayout
+ * in a test infrastructure-specific way.
+ */
+class TextInputAction {
+
+  companion object {
+    /**
+     * Check that the TextInputLayout view has the [expectedErrorText] error text.
+     *
+     * @returns a [BoundedMatcher] that matches the [expectedErrorText] with the
+     * TextInputLayout's error text
+     */
+    fun hasErrorText(expectedErrorText: String): BoundedMatcher<View, TextInputLayout> {
```

Comment:
We always want the argument to be a string from the resources. So, let's annotate this with the `@StringRes`

`fun hasErrorText(@StringRes expectedErrorText: Int)` and then all the functions calling will be look like example below - 
```kt
      onView(withId(R.id.add_profile_activity_user_name))
        .check(matches(hasErrorText(R.string.add_profile_error_name_not_unique)))
```